### ISSUE TITLE
Accept repair status reports from Crucible

### ISSUE THREAD

**Issue by jmpesp** [oxidecomputer/omicron#5120]
Implement the server side of https://github.com/oxidecomputer/crucible/issues/839 by accepting the following notifications:

- live repair has started
- live repair has completed either ok or with an error

**Comment by leftwo** [comment]
Crucible #839 was actually for notifying Nexus that the initial startup of an upstairs required
reconciliation, not `LiveRepair`.

However, the issue is really the same for Reconciliation as it is for LiveRepair, the difference is
just when we are doing the reporting.
Reconciliation: Before the upstairs goes active and starts allowing IOs.
LiveRepair: After the upstairs is active and has already been taking IOs.

I suspect the reporting part will still be the same between the two.

Another part we should also consider is if the Pantry is doing a Repair, that
path can also report back using the same mechanism.

**Comment by jmpesp** [comment]
> I suspect the reporting part will still be the same between the two.

#5135 only accepts live repair status reports. I could also add a PR that accepts reconciliation status reports? It looks like the intention is to provide some sort of progress indication so that it can be exposed to users, otherwise they're just sitting there waiting?

> Another part we should also consider is if the Pantry is doing a Repair, that path can also report back using the same mechanism.

Reports will come from _all_ running Upstairs :)

**Comment by leftwo** [comment]
> > I suspect the reporting part will still be the same between the two.
> 
> #5135 only accepts live repair status reports. I could also add a PR that accepts reconciliation status reports? It looks like the intention is to provide some sort of progress indication so that it can be exposed to users, otherwise they're just sitting there waiting?

Yes. With a notification only the user could know "we are working on it", but ideally some sort of progress notification as a possibility.

> > Another part we should also consider is if the Pantry is doing a Repair, that path can also report back using the same mechanism.
> 
> Reports will come from _all_ running Upstairs :)

Progress reporting for LiveRepair is also probably worth doing, though maybe not quite as important as the user can still use the volume in question while it is being repaired.
